src/pacman/query.c

```diff
@@ -70,6 +73,32 @@ static int search_path(char **filename, struct stat *bufptr)
 
 		if(lstat(fullname, bufptr) == 0) {
 			free(*filename);
+
```

Comment:
Please guard all changes via `#ifdef __MSYS__` so on Linux the original behavior is used.

Reply:
Done.